src/SFA.DAS.Provider.PR.Web.UnitTests/Models/EmployerDetailsModelTests.cs

```diff
@@ -0,0 +1,101 @@
+        });
+    }
+
+    [Test, AutoData]
+    public void ResponseContainsLastRequestOperations_ModelBuildsLastRequestOperationsCorrectly(
+        GetProviderRelationshipResponse response)
+    {
+        response.LastRequestOperations = new Operation[2];
+
+        var actual = (EmployerDetailsViewModel)response;
+
+        Assert.That(actual.LastRequestOperations, Is.EqualTo(response.LastRequestOperations));
+    }
+
+    [Test, AutoData]
+    public void ResponseContainsLastRequestOperations_ModelBuildsHasPermissionsRequestCorrectly(
+        GetProviderRelationshipResponse response)
+    {
+        response.LastRequestOperations = new Operation[2];
+
+        var actual = (EmployerDetailsViewModel)response;
+
+        Assert.That(actual.HasPermissionsRequest, Is.True);
+    }
+
+    [Test]
+    [InlineAutoData(RequestStatus.Sent, PermissionAction.PermissionCreated, EmployerDetailsViewModel.PendingAddTrainingProviderAndPermissionsRequestText)]
+    [InlineAutoData(RequestStatus.Sent, PermissionAction.ApprovalsRelationship, EmployerDetailsViewModel.PendingNotImplementedText)]
+    [InlineAutoData(RequestStatus.New, PermissionAction.PermissionCreated, EmployerDetailsViewModel.NotPendingNotImplementedText)]
+    public void LastActionTextIsSetCorrectly(RequestStatus status, PermissionAction action, string expected,
+        GetProviderRelationshipResponse response)
+    {
+        response.LastRequestStatus = status;
+        response.LastAction = action;
+
+        var actual = (EmployerDetailsViewModel)response;
+
+        Assert.That(actual.LastActionText, Is.EqualTo(expected));
+    }
+
+    [Test, AutoData]
+    public void OperationsContainsCreateCohort_CurrentPermissionsIncludeCreateCohortText(
+        GetProviderRelationshipResponse response)
+    {
+        response.Operations = new Operation[] { Operation.CreateCohort };
+
+        var actual = (EmployerDetailsViewModel)response;
+
+        Assert.That(actual.CurrentPermissions.Contains(EmployerDetailsViewModel.CohortsPermissionText));
+    }
+
+    [Test, AutoData]
+    public void OperationsContainsRecruitment_CurrentPermissionsIncludeRecruitmentText(
+        GetProviderRelationshipResponse response)
+    {
+        response.Operations = new Operation[] { Operation.Recruitment };
+
+        var actual = (EmployerDetailsViewModel)response;
+
+        Assert.That(actual.CurrentPermissions.Contains(EmployerDetailsViewModel.RecruitmentPermissionText));
+    }
+
+    [Test, AutoData]
+    public void OperationsContainsRecruitmentRequiresReview_CurrentPermissionsIncludeRecruitmentRequiresReviewText(
+        GetProviderRelationshipResponse response)
+    {
+        response.Operations = new Operation[] { Operation.RecruitmentRequiresReview };
+
+        var actual = (EmployerDetailsViewModel)response;
+
+        Assert.That(actual.CurrentPermissions.Contains(EmployerDetailsViewModel.RecruitmentRequiresReviewPermissionText));
+    }
```

Comment:
Could merge this in to one test with inline test cases?

Reply:
Merged into one test with inline test cases